backend/test/institution_collection_handler_test.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+"""Institution collection handler test."""
+import operator
+from test_base_handler import TestBaseHandler
+from models.invite_institution import InviteInstitution
+from models.user import User
+from models.institution import Institution
+from handlers.institution_collection_handler import InstitutionCollectionHandler
+from worker import AddAdminPermissionsInInstitutionHierarchy
+from worker import RemoveAdminPermissionsInInstitutionHierarchy
+
+from mock import patch
+import mocks
+
+
+class InstitutionCollectionHandlerTest(TestBaseHandler):
+    """Test the institution_collecion_handler class."""
+
+    @classmethod
+    def setUp(cls):
+        """Provide the base for the tests."""
+        super(InstitutionCollectionHandlerTest, cls).setUp()
+        methods = set(cls.webapp2.WSGIApplication.allowed_methods)
+        methods.add('PATCH')
+        cls.webapp2.WSGIApplication.allowed_methods = frozenset(methods)
+        app = cls.webapp2.WSGIApplication(
+            [("/api/institutions", InstitutionCollectionHandler),
+             ("/api/queue/add-admin-permissions", AddAdminPermissionsInInstitutionHierarchy),
+             ('/api/queue/remove-admin-permissions', RemoveAdminPermissionsInInstitutionHierarchy)
+             ], debug=True)
+        cls.testapp = cls.webtest.TestApp(app)
+        
+        # create models
+        # new User
+        cls.user = mocks.create_user('user@example.com')
+        cls.user.name = "User"
+        # new User Other
+        cls.other_user = mocks.create_user('other_user@example.com')
+        cls.other_user.state = "pending"
+        cls.other_user.put()
+        # new Institution FIRST INST
+        cls.first_inst = mocks.create_institution()
+        cls.first_inst.name = 'FIRST INST'
```

Comment:
You shouldn't do those assignments to the institution model, once is the job of the mock provide a usable resource.